src/nomad_measurements/xrd/schema.py

```diff
@@ -1014,8 +977,12 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger'):
         super().normalize(archive, logger)
         if not self.results:
             return
-        self.write_nx_section_and_create_file(archive, logger)
-        self.figures = self.results[0].generate_plots(archive, logger)
+
+        scan_type = xrd_dict.get('metadata', {}).get('scan_type', None)
+        if self.generate_nexus_file and self.data_file is not None:
+            write_nx_section_and_create_file(archive, logger, scan_type=scan_type)
+
+            self.figures = self.results[0].generate_plots(archive, logger)
```

Comment:
```suggestion
        self.figures = self.results[0].generate_plots(archive, logger)
```

The plotting of figures should not be inside the if block.